internal/contracts/manager.go

```diff
@@ -266,7 +266,9 @@ func (cm *contractManager) BroadcastContractAPI(ctx context.Context, ns string,
 	err = cm.database.RunAsGroup(ctx, func(ctx context.Context) (err error) {
 		existing, err := cm.database.GetContractAPIByName(ctx, api.Namespace, api.Name)
 		if existing != nil && err == nil {
-			return i18n.NewError(ctx, i18n.MsgContractAPIExists, api.Namespace, api.Name)
+			if !existing.Location.Hash().Equals(api.Location.Hash()) || !existing.Ledger.Hash().Equals(api.Ledger.Hash()) {
```

Comment:
Is it worth adding a helper on `ContractAPI` for this check since it's used in at least 2 different places?

Reply:
Yeah that's a good idea